demo/site/src/util/resolveUrl.ts

```diff
@@ -0,0 +1,44 @@
+type ResolveUrlOptions = {
+    /**
+     * the baseUrl will be prepended to the resulting path
+     *
+     * Sample: "http://localhost:3000"
+     * @default: "/"
+     */
+    baseUrl?: string;
+
+    /**
+     * path with leading /
+     *
+     * Sample: "/test/to/my/page"
+     */
+    path: string;
+
+    scope: {
+        language: string;
+    } | null;
+
+    /**
+     * If provided, the anchor will be appended to the resulting URL
+     *
+     * Sample: "my-anchor"
+     */
+    anchor?: string;
+};
+
+/**
+ * Resolves a path, scope and a baseUrl to a real URL which can be navigated to.
+ *
+ * @return {string} The resolved URL: /{scope.language}/test/to/my/page#my-anchor
+ */
+export const resolveUrl = ({ baseUrl = "/", path, scope, anchor }: ResolveUrlOptions) => {
```

Comment:
- baseUrl should not include trailing `/` and default to ""
- I would even consider removing the baseUrl from this new function and keep appending it in sitemap (as this is the only place where it is needed)
- scope.language should be required (in this multi-lang setup) - creating urls without scope.language is not possible and would result in an invalid url

Reply:
renamed `resolveUrl` to `createSiteUrl`

Reply:
I made the scope optional because of the implementation detail of InternalLink: https://github.com/vivid-planet/comet/pull/3099/files#diff-3dfc3d13f8394017033da0b81712accc4475939f36a1df4826a6151807d8ba57R22-R26.

I think scope can be null there, at least its mentioned in the blocks.generated.ts
<img width="429" alt="Screenshot 2025-01-14 at 13 45 45" src="https://github.com/user-attachments/assets/54d5cf4b-7c57-42e1-bcfe-2f4b01714d76" />

I am unsure about that, but I can make the scope required in this function and render the span instead.

Reply:
Concerning the **baseUrl** -> nearly every link I have seen in the demo is starting with a /.

Reply:
but I think it's a good idea to start with default value = empty string.

Reply:
> I think scope can be null there, at least its mentioned in the blocks.generated.ts

The reason why InteranlLinkBlockData["targetPage"]["scope"] is nullalbe, is because it is implemented in the library and the library doesn't know if the application has scoped pages or not. In an concrete application the scope is either always there or not.

The previous implementation is [InternalLinkBlock.tsx](https://github.com/vivid-planet/comet/pull/3099/files#diff-3dfc3d13f8394017033da0b81712accc4475939f36a1df4826a6151807d8ba57) was also wrong: the when `if (targetPage.scope) {` doesn't match you get an invalid link.

I would propose to:
- make createSiteUrl scope argument required
- Render an error (or even throw an error) in InternalLinkBlock if targetPage.scope is null

Reply:
That makes sense, I will adjust this. 👍

Reply:
> Concerning the baseUrl -> nearly every link I have seen in the demo is starting with a /.

true, because we want absolute links. When a link doesn't start with a `/` it is a relative link, relative to the current url. You _never_ want that.

The starting `/` must be part of the url, to state that it is an absolute url.

(see also targetPage.path, that also includes the starting `/`)

Reply:
@nsams done

Reply:
the baseUrl part is unchanged...

Reply:
dropped baseUrl part